drivers/hmis/app/graphql/types/hmis_schema/assessment.rb

```diff
@@ -65,7 +65,7 @@ def definition
       # TODO: check if form is retired? For non-WIP assessments, we should
       # really be choosing the "latest" form, which may not be the one on the FormProcessor.
       form_processor = load_ar_association(object, :form_processor)
-      definition = load_ar_association(form_processor, :definition)
+      definition = load_ar_association(form_processor, :definition) if form_processor
       definition ||= Hmis::Form::Definition.find_definition_for_role(role, project: project)
```

Comment:
> Note, I'm not sure how we get assessments without a nil processor in practice.

Yeah sorry this isn't really explained anywhere. These are the scenarios in which these things occur:
* CustomAssessment with no FormProcessor: this shouldn't happen at this point. I think we are missing some db constraints / validations here.
* CustomAssessment with a FormProcessor _without_ a Definition: this would occur if the assessment was generated from the MigrateAssessmentsJob. That's a migration job that we'll run once (or repeatedly) to generate CustomAssessment records based on data. Basically the way it works is group together all the records that share the same Information Date and Data Collection Stage, and call it an assessment. I suppose we could also attach a definition in that job, but we don't.
* CustomAssessment with a FormProcessor _with_ a Definition: the definition stored is the definition that was most recently used to save or submit the form. The "TODO" on line 65 is indicating that–– once we have a mechanism for retiring "old" versions of forms–– we should probably open this with the "latest" form (Which is returned from find_definition_for_role). I held off on doing that because it's slower, but we'll need to address it somehow once we have an implementation for retiring old form versions.

Reply:
Thanks for the explanation @gigxz! So it sounds like the case I'm handling above (nil processor on assessment) isn't expected. In that case maybe ignoring it isn't the right thing to do and we should raise an exception instead?

Reply:
Yeah you're right, we should - made that change and added some more comments here https://github.com/greenriver/hmis-warehouse/pull/3371